Controls/Chart/Chart.UWP/Visualization/Common/ContainerVisualsFactory.cs

```diff
@@ -266,21 +266,41 @@ protected internal virtual ContainerVisual PrepareCartesianAxisLineVisual(Cartes
         protected internal virtual ContainerVisual PrepareLineRenderVisual(ContainerVisual containerVisual, IEnumerable<Point> points, Brush stroke, double strokeThickness)
         {
             containerVisual.Children.RemoveAll();
-            for (int i = 0; i < points.Count(); i++)
+
+            using (var enumerator = points.GetEnumerator())
             {
-                SpriteVisual childSpiteVisual = containerVisual.Compositor.CreateSpriteVisual();
-                var startPoint = points.ElementAt(i);
-                if (i + 1 < points.Count())
+                var haveStartPoint = false;
+                var startPoint = default(Point);
+
+                while (enumerator.MoveNext())
                 {
-                    var endPoint = points.ElementAt(i + 1);
+                    var endPoint = enumerator.Current;
+
+                    if (!haveStartPoint)
+                    {
+                        startPoint = endPoint;
+                        if (!enumerator.MoveNext()) break;
+                        haveStartPoint = true;
+                        endPoint = enumerator.Current;
+                    }
+
+                    // draw the line if we have both points
+                    var childSpiteVisual = containerVisual.Compositor.CreateSpriteVisual();
                     var angle = Math.Atan2(endPoint.Y - startPoint.Y, endPoint.X - startPoint.X) * (180.0 / Math.PI);
-                    var width = Math.Sqrt((Math.Pow(startPoint.X - endPoint.X, 2) + Math.Pow(startPoint.Y - endPoint.Y, 2)));
+                    var width = Math.Sqrt(Math.Pow(startPoint.X - endPoint.X, 2) + Math.Pow(startPoint.Y - endPoint.Y, 2));
 
-                    childSpiteVisual.RotationAngleInDegrees = (float)angle;
+	                if (Math.Abs(angle) > 40)
+		                width += 2;
+	                else
+		                width++;
+
```

Comment:
Increasing the Width could be part of the whole solution for the issue. However, changing it based on some random value of the angle according to me is not an appropriate solution. What we could start with in order to resolve the issue is to check how the lines are rendered when the orientation of the negative rotated lines is changed - according to me some of the gaps will be filled, but others will remain. Here is a simple code how the rotation could be changed:
`if (angle <= 0)
{
	angle = Math.Abs(angle);
	childSpiteVisual.Orientation = new Quaternion(1, 0, 0, 0);
}`

After that in order to fill the other gaps some calculations might be needed to find the exact size of the gap and change the Widht or even the Offset of the visuals in order to fill it. That could be used as a start point for resolving the whole issue and with your help I believe we will be able to find the right solution.

For now could you please, remove that part of the code with the width. You could remove only that part and made a separate pull request for the fix of the PrepareLineRenderVisual method so, I could be able to merge that pull request.